.github/workflows/test-coverage.yaml

```diff
@@ -15,40 +17,45 @@ jobs:
       GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
 
       - uses: r-lib/actions/setup-r@v2
         with:
           use-public-rspm: true
 
       - uses: r-lib/actions/setup-r-dependencies@v2
         with:
+          extra-packages: any::covr, any::xml2
           needs: coverage
 
-      - name: Setup
-        run: |
-          R CMD INSTALL .
-          source shim_package.sh
-
       - name: Test coverage
         run: |
-          covr::codecov(
+          cov <- covr::package_coverage(
             quiet = FALSE,
             clean = FALSE,
-            install_path = file.path(Sys.getenv("RUNNER_TEMP"), "package")
+            install_path = file.path(normalizePath(Sys.getenv("RUNNER_TEMP"), winslash = "/"), "package")
           )
+          covr::to_cobertura(cov)
         shell: Rscript {0}
 
+      - uses: codecov/codecov-action@v4
```

Comment:
What is the advantage of using the codecov-action vs having covr upload to codecov directly?

Reply:
I don't know 😞 I just updated using our standard scripts. @gaborcsardi might know.

Reply:
covr uses the old protocol, which does not support organization level codecov tokens.

Reply:
🤔

Reply:
Well, you don't have any problems if you don't look at the results. :) Upload failures do not fail the workflow run, so you typically do not notice them:
```
[1] "Rate limit reached. Please upload with the Codecov repository upload token to resolve issue. Expected time to availability: 1300s."
```
https://github.com/r-lib/covr/actions/runs/11729639825/job/32675733827#step:6:88

The last successful upload for covr was ~7 months ago.

Reply:
I seem to no longer have access to the admin settings for this repo, so can't comment on if codecov app is still installed for this repo or the r-lib org in general.

But I don't really like going through the cobutura codepath for repositories by default, it adds some additional complexity and also a xml2 dependency. Additionally the output is really geared towards more java style classes, and doesn't perfectly align with R package organization. Additionally doesn't relying on secrets mean that forks are no longer supported?

I am not sure why the existing path would work differently with organization tokens vs the action? Aren't they both changing the secret into an environment variable in the same way?